### PR TITLE
Added ability to customize ToC

### PR DESCRIPTION
**Pull request by alexstojda**
Implemented "sections" field in the environment file to enable documentation writers to produce more meaningful table of contents groups instead of simply using the tag values.

### CODE REVIEW

**Comment by MikeRalphson** [comment]
Thank you for the PR! Could you take a look at the Travis CI failure, and maybe extract the commit related to schema titles to its own PR, as it seems unrelated to the other changes? 

I do wonder whether it would be easier to control this with a specification extension around the `tags` object rather than adding yet more external configuration. People have previously suggested extensions for sub-tag grouping on the OAS repository. What do you think?

**Comment by alexstojda** [comment]
@MikeRalphson wow you are quick!

I think this type of configuration belongs outside the specification files, because it's not concerned with the functioning of the API, but simply how its displayed in a documentation center. 

I will move the schema title changes to another PR.

**Comment by MikeRalphson** [comment]
> I think this type of configuration belongs outside the specification files, because it's not concerned with the functioning of the API, but simply how its displayed in a documentation center.

But isn't that almost a definition of `tags` in OpenAPI?

**Comment by alexstojda** [comment]
It really depends on the company workflow I guess. For my use case, tags are used to describe a certain functionality that is not necessarily what we want to be represented in the table of contents. Thus having this light-weight functionality would allow a separation of concerns in that regard. 

Plus there are some cases where the tag name and/or description are not what you want as list items in the table of contents. Since OAS allows markdown in its tag descriptions, those may be too long or too descriptive for the ToC.

**Comment by MikeRalphson** [comment]
Something's still niggling me about this PR and it might just be the name `sections`. Perhaps `tagGroups` would be clearer? 

Any other user feedback would be appreciated.

**Comment by maciejsaw** [comment]
Hi, just wanted to mention that we also need this feature. Thank you @alexstojda for this pull request.
I was thinking that it might also be a parameter similar to ```--summary``` to use the tag description as TOC display name, for example ```--use-tag-description```

**Comment by alexstojda** [comment]
@maciejsaw Thanks for the support! 
@MikeRalphson I did the requested refactoring :)

File: openapi3.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ function convertToToc(source,data) {
                 method.slug = sMethodUniqueName.toLowerCase().split(' ').join('-'); // TODO {, } and : ?
                 var tagName = data.translations.defaultTag;
                 if (method.operation.tags && method.operation.tags.length > 0) {
-                    tagName = getSectionTag(method.operation.tags[0], data.options.sections);
+                    tagName = getTagGroup(method.operation.tags[0], data.options.sections);
```
Comment (MikeRalphson) [inline openapi3.js]: Sorry to be a total pain, but could we rename `options.sections` too?

Comment (alexstojda) [reply openapi3.js]: Done! Good catch, it should have been changed as well.

**Comment by MikeRalphson** [comment]
Released as part of `v3.5.0`, thanks again.

**Comment by alexstojda** [comment]
Thank you @MikeRalphson !

**Pull request merged**